.stylelintrc.json

```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "stylelint-config-sass-guidelines",
+  "rules": {
+    "selector-class-pattern": [
+      "^[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*(?:__[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*){0,1}(?:(?:--)[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*){0,1}$",
+      {
+        "resolveNestedSelectors": true,
+        "message": "Expected class selector to match BEM naming convention"
```

Comment:
I tried to keep the message similar to other messages that stylelint have. Because it is custom, we cannot see the class name in the message. It has an issue in [their github](https://github.com/stylelint/stylelint/issues/4117).

Reply:
Nice!!